### PR TITLE
creating install.json file related to apm single step instrumation

### PR DESCRIPTION
**Pull request by schneideradam**
### What does this PR do?

This PR makes the agent install ansible collection write an `install.json` file to the same directory as the Agent config file, `datadog.yaml`. The `install.json` file contains a random UUID, an installation timestamp, and the mechanism that was used to install the agent (the hardcoded string `ansible_manual` or `ansible_single_step`). The agent will send this information to the Datadog backend, so that we can gather telemetric data on how long users take to onboard after the initial installation of the agent.

### Motivation

Jira: https://datadoghq.atlassian.net/browse/APMON-1070 . This is part of our broader work on improving the APM onboarding experience.

### Possible Drawbacks / Trade-offs

I am not copying these changes to the `agent-dmg`, `iot-agent-deb` or `iot-agent-rpm` directories, because we are initially focusing on improving the onboarding experience on Linux.

## Quick test
There isn't a `quick test` because of the nature of the ansible installer. Please follow the non-quick test below.

## End to end test
To test, I first spun up a gcp RHEL instance and pulled the latest branch code. I installed ansible-core v2.11.12.  I confirmed that the `datadog_apm_instrumentation_enabled` was set to `"host"`.
From there I ran the playbook with the manual test listed in the command below:

```
ansible-playbook -i localhost, manual_tests/test_7_full.yml --connection=local
```

I checked in `/etc/datadog-agent` to ensure install.json existed and had the following values:
`{"install_id": "88d5cfef-2f72-4b59-af8d-1e07b71b684b", "install_type": "ansible_single_step", "install_time": 1718116519}`

I then changed the value for `datadog_apm_instrumentation_enabled` to an empty string `""` and re-ran the command. I observed that the following install.json updated with the following:

`{"install_id": "88d5cfef-2f72-4b59-af8d-1e07b71b684b", "install_type": "ansible_manual", "install_time": 1718116519}`

I repeated all above steps in debian.

### CODE REVIEW

File: tasks/_agent-linux-macos-shared.yml [hunk 1]
```diff
@@ -97,3 +97,50 @@
...
+    - name: Generate uuid using uuidgen
+      command: uuidgen
+      register: uuid_cmd
+      changed_when: false
+
+- name: Set install signature from file if it exists
+  set_fact:
+    install_signature:
+      install_id: "{{ install_info.install_id }}"
+      install_time: "{{ install_info.install_time }}"
```
Comment (yshapiro-57) [inline tasks/_agent-linux-macos-shared.yml:130]: I am sorry, should this also include an install_type?

File: tasks/_agent-linux-macos-shared.yml [hunk 2]
```diff
@@ -97,3 +97,50 @@
...
+  when: install_file.stat.exists
+
+- name: Generate install signature
+  set_fact:
+    install_signature:
+      install_id: "{{ uuid_cmd.stdout }}"
+      install_time: "{{ ansible_date_time.epoch }}"
```
Comment (yshapiro-57) [inline tasks/_agent-linux-macos-shared.yml:137]: Same here, should this include an install type in addition to install ID and time?

**Review by yshapiro-57 — COMMENTED** [review]
Is there any test plan that can be run to install the agent on Ansible with this code in place and verify that an install.json file gets generated?

File: tasks/_agent-linux-macos-shared.yml (hunk 1 above)
Comment (schneideradam) [reply tasks/_agent-linux-macos-shared.yml:130]: install_type is picked up in the .j2 template.

File: tasks/_agent-linux-macos-shared.yml (hunk 2 above)
Comment (schneideradam) [reply tasks/_agent-linux-macos-shared.yml:137]: install_type is picked up in the .j2 template.

**Comment by schneideradam** [comment]
I have verified this on both rhel and debian. I checked with both `datadog_apm_instrumentation_enabled` enabled, and disabled, and verified the correct behavior. Let me know if that is adequate. I can share screenshots if you'd like.

**Review by yshapiro-57 — APPROVED** [review]
@schneideradam I trust you, you don't have to upload screenshots. I am just a fan of leaving test plans on PRs so that my successor knows how to maintain my code after I get promoted to VP and no longer have time to deal with this stuff.

Thank you very much again for helping me and @yahya-mouman on short notice with these changes. This was very nice of you.

**Comment by schneideradam** [comment]
@yshapiro-57 Do you mean you'd like to see some sort of CI unit test? Or instructions on how to test this PR? I'm happy to do whatever, I just usually don't worry about testing something like this when it should just fail silently so as not to get in the way of instrumenting their applications.

**Comment by yshapiro-57** [comment]
@schneideradam I was thinking [something like this wiki](https://datadoghq.atlassian.net/wiki/spaces/AO/pages/2983035648/Cluster+Agent+Development) or [the end to end test plan on this PR](https://github.com/DataDog/datadog-agent/pull/23084); something that explains to a fresh person how to cause this code to be executed in a local environment.

Again, if you don't already have a writeup handy, and putting one together is time-consuming, then don't worry about it. I think it will be fine to document the testing steps on RHEL and Debian another time.

**Review by spencergilbert — APPROVED** [review]

**Review by yshapiro-57 — APPROVED** [review]

**Comment by schneideradam** [comment]
@spencergilbert Can we get this merged? Also, what is the process for deploying to ansible-galaxy?

**Comment by spencergilbert** [comment]
Followed up internally regarding release, expected publishing next week.

**Comment by Kyle-Verhoog** [comment]
@spencergilbert did the release go out?

cc @alexbryukhanova

**Pull request merged**